File: Swift/0011.swift
class Solution {
    func maxArea(_ height: [Int]) -> Int {
        var left = 0
        var right = height.count - 1
        var maxArea = 0

        while left < right {
            let width = right - left
            let currentHeight = min(height[left], height[right])
            let area = width * currentHeight
            maxArea = max(maxArea, area)

            // Move the pointer pointing to the shorter line
            if height[left] < height[right] {
                left += 1
            } else {
                right -= 1
            }
        }

        return maxArea
    }
}

File: Rust/0011.rs
impl Solution {
    pub fn max_area(height: Vec<i32>) -> i32 {
        let mut left = 0;
        let mut right = height.len() - 1;
        let mut max_area = 0;

        while left < right {
            let width = (right - left) as i32;
            let current_height = height[left].min(height[right]);
            let area = width * current_height;
            max_area = max_area.max(area);

            // Move the pointer pointing to the shorter line
            if height[left] < height[right] {
                left += 1;
            } else {
                right -= 1;
            }
        }

        max_area
    }
}
